torch/_decomp/decompositions_for_jvp.py

```diff
@@ -1,17 +1,31 @@
+import inspect
+from typing import Callable, Dict, List, Optional, Tuple
+
 import torch
-from torch import Tensor
 import torch._decomp
-from typing import Tuple, List, Optional
-
-aten = torch.ops.aten
+from torch import Tensor
 
 decomposition_table = torch._decomp.decomposition_table
+decomposition_table_for_jvp: Dict[torch._ops.OpOverload, Callable] = {}
 register_decomposition = torch._decomp.register_decomposition
-get_decompositions = torch._decomp.get_decompositions
+aten = torch.ops.aten
 
-# Decompositions have been ported to torch._decomp inside of PyTorch core.
-# The only decompositions here are temporary or hacks.
-# Please submit your contributions to PyTorch core!
+# NOTE: [forward-mode AD decompositions hack]
+#
+# The mechanism is in VariableType,
+#   IF any inputs have forward grad
+#      AND there is no forward AD formula implemented
+#      AND the functions is actually differentiable
+#   run the decomposition
+#      See run_jit_decomposition_with_args_for_jvp
+#      We currently use python decompositions that we torchscript.
+#
+# Note that we would be building the backward graph at the decomposed level
+# too, but that is OK, because we would've errored out otherwise anyway.
+#
+# TODO: what if jit decompositions exists, should we just use it?
+#       or do we want to have an explicit white list like functorch had
+#       using special JVP_DECOMP DynamicLayerFront kernel
```

Comment:
This probably belongs in the next PR of the stack, but happy to keep it here too

Reply:
Yeah either's fine to me